### DISCUSSION TITLE
How to add New CONNECTIONS type

### DISCUSSION THREAD

**Discussion by unanao** [microsoft/promptflow#996] [Q&A] [answered]
I can add new connection by "Custom" type.  As the key value is not specified,  therefore it is a bit of inconvenience。 Could I add our own  CONNECTION type just like "Azure OpenAI" or Qdrant?

**Comment by thy09** [comment, marked as answer]
You may check this document: https://microsoft.github.io/promptflow/how-to-guides/develop-a-tool/create-your-own-custom-strong-type-connection.html

**Reply by unanao** [reply]
Thanks!  It's useful a lot. Can I put the "Custom connection" under "CONNECTION", not under "Custom"?

**Reply by 16oeahr** [reply]
Currently, it's not supported in this manner. Custom-defined connections are closely linked with the built-in CustomConnection type. In the storage, these custom-defined connections are saved as a custom connection. 
In local extension, you can distinguish the custom-defined connections by the type hint in gray.
![image](https://github.com/microsoft/promptflow/assets/46446115/c284d7c5-8d05-4f80-a5d2-62d1832c53a7)

Could you provide more details on the need for separating them?

**Reply by unanao** [reply]
Thanks.  As  only use our own LLM, the pre-defined connections is no use for. If we can custom the "CONNECTIONS" list, It would be more convenient for us.   I think the lots of company just like us .

If we change the promptflow code, will bring the trouble of merging code

**Reply by JYC-99** [reply]
Regarding the ask to hide built-in connection type groups to make it less annoying to people who does not need them, we will trigger an internal loop to improve the experience. Probably a global setting option to hide them or a simple toggle button to hide built-ins with one click. Hopefully that will address your ask to customize the connections list.

**Reply by unanao** [reply]
Perfect！ Thanks！

**Reply by VCSinc** [reply]
I have a unique use case where I already have an LLM project deployed using Langchain and several other modules, but I would like to secure the API Key and Endpoint used from Azure. I was hoping to use the "Prompt Flow Connection" to provide a secure tunnel for accessing the Endpoint and API Key since I'm already using Azure OpenAI models.

Is it possible to use the "Connection" created in the "Microsoft Azure ML Studio" directly in VS Code using the "Prompt Flow" module? See the screenshot for the "Connection" in "Azure AI Machine Learning Studio". 

 Is it possible to create a connection.yaml file to make the link to the end point within the Azure network? 

Sorry if this is confusing or already answered.
<img width="608" alt="promptflow" src="https://github.com/microsoft/promptflow/assets/149602802/5e50b4cd-3d2e-4ab2-8c5c-ba8c441d2c45">

**Reply by brynn-code** [reply]
Yes, of course, sorry for the late response, @VCSinc, please see docs [here](https://microsoft.github.io/promptflow/how-to-guides/set-global-configs.html#set-config) to set connection provider to azure workspace, after that promptflow is able to use workspace connections.
